### ISSUE TITLE
Switch from mocking to actual databases/deployments

### ISSUE THREAD

**Issue by spbnick** [kernelci/kcidb#377]
Now that we have fixtures for actual databases and deployments (in `conftest.py`), switch the `test_*_main` tests using `Mock` across the project to using the fixtures instead. Use `test_query` and `test_email_generated` for reference.

However, exercise caution when using deployment fixture, as those take quite a while to cleanup, and too many tests using that would make the overall run even longer. Alternatively, consider introducing a "half-way cleaned" fixture for tests requiring only a part of the deployment, which would take a shorter cleanup.

**Comment by octonawish-akcodes** [comment]
Can I work on this one.

**Comment by spbnick** [comment]
Removing @octonawish-akcodes from this issue, as he'd like to pick another one instead.

**Comment by spbnick** [comment]
A good start on this issue would be to start with tests that only require a database (either "clean" or "empty"), since those don't require access to Google Cloud.

**Comment by NikkyXO** [comment]
okay..i can try working on this

**Comment by spbnick** [comment]
Great! Assigning :)

**Comment by danghai** [comment]
I am very happy to see Nick still mentoring many people. I am one of the people that @spbnick has helped a lot during my days of learning to code and contributing to open source. I believe that Nick is one of the best teachers in the world that I admire. I have learned a lot from him from the first commits with very detailed and meticulous comments and feedback. I believe that with Nick's help, you will become great engineers in the future. Thank you Nick again. 👍

**Comment by NikkyXO** [comment]
> I am very happy to see Nick still mentoring many people. I am one of the people that @spbnick has helped a lot during my days of learning to code and contributing to open source. I believe that Nick is one of the best teachers in the world that I admire. I have learned a lot from him from the first commits with very detailed and meticulous comments and feedback. I believe that with Nick's help, you will become great engineers in the future. Thank you Nick again. 👍 

Yeah. He is very good teacher 😊.  He is quite inspiring and seems to know a lot. I really do hope that one day, I can know as much he does. I'm also getting quite comfortable with my git commits. Hoping to make good use of this learning opportunity
